From 7b614cb22641de2ea96cbdadb4e2a93699fe2381 Mon Sep 17 00:00:00 2001
From: Huang Rui <vowstar@gmail.com>
Date: Sat, 26 Apr 2025 15:50:48 +0800
Subject: [PATCH] fix(crypto): Disable SRTP support in mediastreamer2

- Replace SRTP key setting functions with error messages
- Update function signatures to use const uint8_t* for key params
- Add const qualifier to MSEKTParametersSet parameter in set_ekt

Signed-off-by: Huang Rui <vowstar@gmail.com>
---
 src/crypto/ms_srtp.cpp | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/src/crypto/ms_srtp.cpp b/src/crypto/ms_srtp.cpp
index 88380482..da38b6d4 100644
--- a/src/crypto/ms_srtp.cpp
+++ b/src/crypto/ms_srtp.cpp
@@ -1476,7 +1476,8 @@ extern "C" int ms_media_stream_sessions_set_srtp_recv_key(MSMediaStreamSessions
                                                           const uint8_t *key,
                                                           size_t key_length,
                                                           MSSrtpKeySource source) {
-	return ms_media_stream_sessions_set_srtp_key(sessions, suite, key, key_length, false, false, source);
+	ms_error("Unable to set srtp recv key: srtp support disabled in mediastreamer2");
+	return -1;
 }
 
 extern "C" int ms_media_stream_sessions_set_srtp_inner_recv_key(MSMediaStreamSessions *sessions,
@@ -1493,7 +1494,8 @@ extern "C" int ms_media_stream_sessions_set_srtp_send_key(MSMediaStreamSessions
                                                           const uint8_t *key,
                                                           size_t key_length,
                                                           MSSrtpKeySource source) {
-	return ms_media_stream_sessions_set_srtp_key(sessions, suite, key, key_length, true, false, source);
+	ms_error("Unable to set srtp send key: srtp support disabled in mediastreamer2");
+	return -1;
 }
 
 extern "C" int ms_media_stream_sessions_set_srtp_inner_send_key(MSMediaStreamSessions *sessions,
@@ -1634,7 +1636,7 @@ extern "C" int ms_media_stream_sessions_set_srtp_recv_key_b64(struct _MSMediaStr
 }
 
 extern "C" int ms_media_stream_sessions_set_srtp_recv_key(
-    MSMediaStreamSessions *sessions, MSCryptoSuite suite, const char *key, size_t key_length, MSSrtpKeySource source) {
+    MSMediaStreamSessions *sessions, MSCryptoSuite suite, const uint8_t *key, size_t key_length, MSSrtpKeySource source) {
 	ms_error("Unable to set srtp recv key: srtp support disabled in mediastreamer2");
 	return -1;
 }
@@ -1648,7 +1650,7 @@ extern "C" int ms_media_stream_sessions_set_srtp_send_key_b64(MSMediaStreamSessi
 }
 
 extern "C" int ms_media_stream_sessions_set_srtp_send_key(
-    MSMediaStreamSessions *sessions, MSCryptoSuite suite, const char *key, size_t key_length, MSSrtpKeySource source) {
+    MSMediaStreamSessions *sessions, MSCryptoSuite suite, const uint8_t *key, size_t key_length, MSSrtpKeySource source) {
 	ms_error("Unable to set srtp send key: srtp support disabled in mediastreamer2");
 	return -1;
 }
@@ -1691,7 +1693,7 @@ extern "C" int ms_media_stream_sessions_set_ekt_mode(MSMediaStreamSessions *sess
 	ms_error("Unable to set EKT operation mode: srtp support disabled in mediastreamer2");
 	return -1;
 }
-extern "C" int ms_media_stream_sessions_set_ekt(MSMediaStreamSessions *sessions, MSEKTParametersSet *ekt) {
+extern "C" int ms_media_stream_sessions_set_ekt(MSMediaStreamSessions *sessions, const MSEKTParametersSet *ekt) {
 	ms_error("Unable to set EKT key: srtp support disabled in mediastreamer2");
 	return -1;
 }
-- 
2.49.0

